src/main/scala/codacy/tailor/Tailor.scala

```diff
@@ -35,69 +30,95 @@ object Tailor extends Tool {
-        "--") ++ filesToLint
-
-      CommandRunner.exec(command) match {
-        case Right(resultFromTool) =>
-          parseToolResult(Paths.get(source.path), resultFromTool.stdout) match {
-            case s @ Success(_) => s
-            case Failure(e) =>
-              val msg =
-                s"""
-                   |${this.getClass.getSimpleName} exited with code ${resultFromTool.exitCode}
-                   |command: ${command.mkString(" ")}
-                   |message: ${e.getMessage}
-                   |stdout: ${resultFromTool.stdout.mkString(
-                     Properties.lineSeparator)}
-                   |stderr: ${resultFromTool.stderr.mkString(
-                     Properties.lineSeparator)}
+  private def failureMessage(command: List[String], resultFromTool: CommandResult, e: Throwable): String = {
+    s"""
+       |${this.getClass.getSimpleName} exited with code ${resultFromTool.exitCode}
+       |command: ${command.mkString(" ")}
+       |message: ${e.getMessage}
+       |stdout: ${resultFromTool.stdout.mkString(Properties.lineSeparator)}
+       |stderr: ${resultFromTool.stderr.mkString(Properties.lineSeparator)}
                 """.stripMargin
-              Failure(new Exception(msg))
-          }
+  }
 
-        case Left(e) =>
-          Failure(e)
-      }
+  def listOfFilesToLint(source: Source.Directory, files: Option[Set[Source.File]]): List[String] = {
+    files.fold(List(source.path)) { paths =>
+      paths.map(_.toString).toList
+    }
+  }
+
+  def commandToRun(patternsConfig: List[String], filesToLint: List[String]): List[String] = {
+    List("/usr/bin/tailor/bin/tailor", "-f", "json") ++ patternsConfig ++ List("--") ++ filesToLint
+  }
+
+  def runCommandOnSourceDir(command: List[String], source: Source.Directory): Try[List[Result]] = {
+    CommandRunner.exec(command) match {
+      case Right(resultFromTool) =>
+        parseToolResult(Paths.get(source.path), resultFromTool.stdout) match {
+          case s @ Success(_) => s
+          case Failure(e) =>
+            val msg = failureMessage(command, resultFromTool, e)
+            Failure(new Exception(msg))
+        }
+
+      case Left(e) =>
+        Failure(e)
+    }
+  }
+
+  override def apply(
+      source: Source.Directory,
+      configuration: Option[List[Pattern.Definition]],
+      files: Option[Set[Source.File]],
+      options: Map[Options.Key, Options.Value]
+  )(implicit specification: Specification): Try[List[Result]] = {
+    Try {
+      val filesToLint: List[String] = listOfFilesToLint(source, files)
+
+      val patternsConfig = lintingConfiguration(source, configuration)
+
+      val command = commandToRun(patternsConfig, filesToLint)
+
+      runCommandOnSourceDir(command, source)
     }.flatten
   }
 
-  private def parseToolResult(path: Path,
-                              output: List[String]): Try[List[Result]] = {
+  private def parseToolResult(path: Path, output: List[String]): Try[List[Result]] = {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [parseToolResult should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)